src/main/proto/com/google/api/codegen/config.proto

```diff
@@ -292,6 +300,84 @@ message MethodConfigProto {
 
 }
 
+// `CodeSampleValues` defines sets of parameter values to be used in
+// generating samples.
+message CodeSampleValues {
+  // An identifier, unique to this method, for this set of values.
+  string id = 1;
+
+  // A short, user-visible name for this set of values. This name may
+  // be used in a UI menu, for example.
+  string title = 2;
+
+  // A longer description for this set of values. This name may be
+  // used in a UI hoverbox, for example.
+  string description = 3;
+
+  // A set of parameter values to be used together in samples. Each
+  // member of `parameters` can be specified in one of two formats:
+  //
+  // * `DOTTED.RESOURCE.NAME=VALUE`, to specify the entire value of
+  //   `NAME`.  If `VALUE` is not specified, it is treated as the
+  //    default zero value for `NAME`'s type.
+  // *  `DOTTED.RESOURCE.NAME#FIELD=VALUE`, if `NAME` is required to be
+  //    a specially formatted string using `FIELD` as one of possibly
+  //    several placeholders.
+  repeated string parameters = 4;
+}
+
+// `SampleConfigurations` defines combinations of "molds" (sample code
```

Comment:
Shouldn't `SampleConfigurations` be singular to match the message name?

Reply:
Good catch!